src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.ValueTaskSource.cs

```diff
@@ -29,37 +29,40 @@ private unsafe class ValueTaskSource : IValueTaskSource<int>, IValueTaskSource
             private bool _cancellationHasBeenRegistered;
 #endif
 
-            public static ValueTaskSource Create(
-                AsyncWindowsFileStreamStrategy strategy,
-                PreAllocatedOverlapped? preallocatedOverlapped,
-                ReadOnlyMemory<byte> memory)
+            internal ValueTaskSource(AsyncWindowsFileStreamStrategy strategy)
             {
-                // If the memory passed in is the strategy's internal buffer, we can use the base AwaitableProvider,
-                // which has a PreAllocatedOverlapped with the memory already pinned.  Otherwise, we use the derived
-                // MemoryAwaitableProvider, which Retains the memory, which will result in less pinning in the case
-                // where the underlying memory is backed by pre-pinned buffers.
-                return preallocatedOverlapped != null &&
-                       MemoryMarshal.TryGetArray(memory, out ArraySegment<byte> buffer) &&
-                       preallocatedOverlapped.IsUserObject(buffer.Array) ?
-                            new ValueTaskSource(strategy, preallocatedOverlapped, buffer.Array) :
-                            new MemoryValueTaskSource(strategy, memory);
+                _strategy = strategy;
+                _result = TaskSourceCodes.NoResult;
+                _source = default;
+                _source.RunContinuationsAsynchronously = true;
             }
 
-            protected ValueTaskSource(
-                AsyncWindowsFileStreamStrategy strategy,
-                PreAllocatedOverlapped? preallocatedOverlapped,
-                byte[]? bytes)
+            internal ValueTaskSource(AsyncWindowsFileStreamStrategy strategy, byte[] internalBuffer) : this(strategy)
             {
                 _strategy = strategy;
                 _result = TaskSourceCodes.NoResult;
-
                 _source = default;
                 _source.RunContinuationsAsynchronously = true;
```

Comment:
If you're delegating to the other ctor with `: this(strategy)`, the above assignments are all duplicative.